cpp/csp/adapters/websockets/ClientAdapterManager.cpp

```diff
@@ -0,0 +1,139 @@
+#include <csp/adapters/websockets/ClientAdapterManager.h>
+
+#include <csp/core/Platform.h>
+#include <chrono>
+#include <iomanip>
+#include <iostream>
+
+
+using websocketpp::lib::placeholders::_1;
+using websocketpp::lib::placeholders::_2;
+using websocketpp::lib::bind;
+
+namespace csp {
+
+INIT_CSP_ENUM( adapters::websockets::ClientStatusType,
+               "ACTIVE",
+               "GENERIC_ERROR",
+               "CONNECTION_FAILED",
+               "CLOSED",
+);
+
+}
+
+// With TLS
+namespace csp::adapters::websockets {
+
+ClientAdapterManager::ClientAdapterManager( Engine* engine, const Dictionary & properties 
+) : AdapterManager( engine ), 
+    m_active(false), 
+    m_shouldRun(false), 
+    m_endpoint(nullptr),
+    m_inputAdapter(nullptr), 
+    m_outputAdapter(nullptr),
+    m_updateAdapter(nullptr),
+    m_thread(nullptr), 
+    m_properties(properties) 
+{
+    if (m_properties.get<bool>("use_tls")) {
+        m_endpoint = new WebsocketEndpointTLS(properties);
+    } else {
+        m_endpoint = new WebsocketEndpointNoTLS(properties);
+    }
+
+    m_endpoint->setOnMessageCb(std::move([this](std::string msg) {
```

Comment:
would it make more sense to check m_inputAdapter != null before setting the CB instead?